### PR TITLE
Update Github Actions to use the Build Chain

### PR DESCRIPTION
**Pull request by radtriste**
### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
</details>

### CODE REVIEW

**Comment by danielezonca** [comment]
@radtriste 
Can you please apply the same fix of this PR?
https://github.com/kiegroup/drools/pull/3642/files
Otherwise the risk is to mix m2 created with different jdk versions

File: .github/workflows/os_java_ci.yml [hunk 1]
```diff
@@ -21,32 +21,77 @@ jobs:
...
+          definition-file: https://raw.githubusercontent.com/${GROUP}/kogito-pipelines/${BRANCH}/.ci/pull-request-config.yaml
+          starting-project: kiegroup/optaplanner
+      - name: Publish Test Report
+        if: ${{ always() }}
+        uses: ginxo/action-surefire-report@BXMSPROD-1311
+        with:
+          fail_on_test_failures: true
+          fail_if_no_tests: false
+          skip_check_run_creation: true
+          report_paths: '**/*-reports/TEST-*.xml'
```
Comment (danielezonca) [inline .github/workflows/os_java_ci.yml:69]:
```suggestion
        uses: ScaCap/action-surefire-report@v1.0.10
        with:
          fail_on_test_failures: true
          fail_if_no_tests: false
          skip_publishing: true
          report_paths: '**/*-reports/TEST-*.xml'
```

File: .github/workflows/os_java_ci.yml [hunk 2]
```diff
@@ -56,7 +101,19 @@ jobs:
...
```
Comment (danielezonca) [inline .github/workflows/os_java_ci.yml:119]:
```suggestion
        uses: ScaCap/action-surefire-report@v1.0.10
        with:
          fail_on_test_failures: true
          fail_if_no_tests: false
          skip_publishing: true
          report_paths: '**/*-reports/TEST-*.xml'
```

File: .github/workflows/os_java_ci.yml [hunk 3]
```diff
@@ -56,7 +104,19 @@ jobs:
...
-        run: mvn -B --fail-at-end clean install -Dfull
+      - name: Build Chain 16
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@v2.5.7
+        with:
+          definition-file: https://raw.githubusercontent.com/${GROUP}/kogito-pipelines/${BRANCH}/.ci/pull-request-config.yaml
+          starting-project: kiegroup/optaplanner
+      - name: Publish Test Report
+        if: ${{ always() }}
+        uses: ginxo/action-surefire-report@BXMSPROD-1311
```
Comment (Ginxo) [inline .github/workflows/os_java_ci.yml:117]: please replace `Publish Test Report` by

```
      - name: Check Surefire Report
        if: ${{ always() }}
        uses: ScaCap/action-surefire-report@v1.0.10
        with:
          fail_on_test_failures: true
          fail_if_no_tests: false
          skip_publishing: true.
          report_paths: '**/*-reports/TEST-*.xml'
```

File: .github/workflows/os_java_ci.yml [hunk 4]
```diff
@@ -20,33 +20,81 @@ jobs:
...
           restore-keys: ${{ runner.os }}-m2
-      - name: Build with Maven
-        run: mvn -B --fail-at-end clean install -Dfull
+      - name: Build Chain ${{ matrix.java-version }}
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@v2.5.7
+        with:
+          definition-file: https://raw.githubusercontent.com/${GROUP}/kogito-pipelines/${BRANCH}/.ci/pull-request-config.yaml
+          starting-project: kiegroup/optaplanner
+      - name: Publish Test Report
```
Comment (Ginxo) [inline .github/workflows/os_java_ci.yml:65]: please replace `Publish Test Report` by

```
      - name: Check Surefire Report
        if: ${{ always() }}
        uses: ScaCap/action-surefire-report@v1.0.10
        with:
          fail_on_test_failures: true
          fail_if_no_tests: false
          skip_publishing: true.
          report_paths: '**/*-reports/TEST-*.xml'
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
please change `Publish Test Report` step for both flows since changes from `ginxo/action-surefire-report@BXMSPROD-1311` has been merged to original project

File: .github/workflows/os_java_ci.yml (hunk 3 above)
Comment (radtriste) [reply .github/workflows/os_java_ci.yml]: updated

File: .github/workflows/os_java_ci.yml (hunk 4 above)
Comment (radtriste) [reply .github/workflows/os_java_ci.yml]: updated

**Comment by radtriste** [comment]
@Ginxo Can you check it again ? :)

**Comment by rsynek** [comment]
@radtriste there are growing concerns about the limits we are hitting with GHA. As @Ginxo kindly offered to organize a session about our long-term plans with GHA and Jenkins, could we please wait with merging this PR for the outcomes of the discussion? I assume you will be among the invitees.

**Comment by radtriste** [comment]
@rsynek Sure. I was just harmonizing the build on GHAction as there was no "installation" of upstream repos (like kogito-runtimes) and so the PR check needed to wait for the upload of artifacts by the nightly else it was failing...

**Comment by triceo** [comment]
I do not have an issue with this. The number of jobs stays the same. The runtime will grow slightly, but the benefit of no longer having to wait for nightly SNAPSHOTs outweighs that I think.

**Comment by radtriste** [comment]
Indeed dependency is only on kogito-runtimes: https://github.com/kiegroup/kogito-pipelines/blob/master/.ci/project-dependencies.yaml#L5
and the upstream command is https://github.com/kiegroup/kogito-pipelines/blob/master/.ci/pull-request-config.yaml#L11
which takes 1 to 2 minutes on kogito-runtimes

**Review by rsynek — APPROVED** [review]

**Comment by rsynek** [comment]
@radtriste the build of `kogito-runtimes` without tests is truly quick; please disregard my previous comment.

**Review by Ginxo — APPROVED** [review]
LGTM, thanks

**Review by danielezonca — APPROVED** [review]

**Pull request merged**